src/main/java/io/airlift/slice/Slice.java

```diff
@@ -58,6 +59,12 @@
 public final class Slice
         implements Comparable<Slice>
 {
+    static {
+        if (ByteOrder.nativeOrder() != ByteOrder.LITTLE_ENDIAN) {
+            throw new IllegalStateException("Slice requires a little endian platform");
```

Comment:
This is not the right exception, since we're not in a method that is being invoked at the inappropriate time. We could use `RuntimeException` or `Error` here.

Reply:
Yeah, actually an AssertionError might be more appropriate.